funnel/models/project.py

```diff
@@ -210,30 +214,34 @@ def datelocation(self):
         If multi-day event across years
         > 30 Dec 2018–02 Jan 2019, Bangalore
 
-        ``datelocation_format`` always keeps ``date_upto`` format as ``–DD Mmm YYYY``.
-        Depending on the scenario mentioned below, format for ``date`` changes. Above examples
+        ``datelocation_format`` always keeps ``schedule_end_at`` format as ``–DD Mmm YYYY``.
+        Depending on the scenario mentioned below, format for ``schedule_start_at`` changes. Above examples
         demonstrate the same. All the possible outputs end with ``–DD Mmm YYYY, Venue``.
-        Only ``date`` format changes.
+        Only ``schedule_start_at`` format changes.
         """
-        datelocation_format = u"{date}–{date_upto} {year}"
-        if self.date == self.date_upto:
-            # if both dates are same, in case of single day project
-            strf_date = ""
-            datelocation_format = u"{date_upto} {year}"
-        elif self.date.month == self.date_upto.month:
-            # If multi-day event in same month
-            strf_date = "%d"
-        elif self.date.month != self.date_upto.month:
-            # If multi-day event across months
-            strf_date = "%d %b"
-        elif self.date.year != self.date_upto.year:
-            # if the start date and end dates are in different years,
-            strf_date = "%d %b %Y"
-        datelocation = datelocation_format.format(
-            date=self.date.strftime(strf_date),
-            date_upto=self.date_upto.strftime("%d %b"),
-            year=self.date.year)
-        return datelocation if not self.location else u', '.join([datelocation, self.location])
+        daterange = u""
+        if self.schedule_start_at and self.schedule_end_at:
```

Comment:
Don't need to test for both, since sessions already enforce that `start_at` and `end_at` must exist together. If there is a `schedule_start_at`, then `schedule_end_at` is also guaranteed to exist.

Reply:
However, the test is harmless as both properties are cached. **But** you need to test with `is not None` because `date(0, 0, 0)` is falsy.